scripts/providers/import_wikisource.py

```diff
@@ -0,0 +1,827 @@
+                    mwapi:gcmtitle "'''
+            + self._catformat(category)
+            + '''".
+    ?page wikibase:apiOutput mwapi:title.
+    ?item wikibase:apiOutputItem mwapi:item .
+  }
+
+  ?item wdt:P31/wdt:P279* ?instanceOf.
+  '''
+            + ''.join(
+                [
+                    f"FILTER NOT EXISTS {{ ?item wdt:P31/wdt:P279* wd:{type}. }}\n  "
+                    for type in EXCLUDED_WIKIDATA_SUBCLASSES
+                ]
+            )
+            + ''.join(
+                [
+                    f"FILTER NOT EXISTS {{ ?item wdt:P31 wd:{type}. }}\n  "
+                    for type in EXCLUDED_WIKIDATA_INSTANCES
+                ]
+            )
+            + ''.join(
+                [
+                    f"FILTER NOT EXISTS {{ ?item wdt:P136 wd:{type}. }}\n  "
+                    for type in EXCLUDED_WIKIDATA_GENRES
+                ]
+            )
+            + '''
+  FILTER (!CONTAINS(STR(?page), "/"))
+  SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE],mul,'''
+            + self.langcode
+            + '''". }
+}'''
+        )
+
+    def _sparql_url(self, category: str) -> str:
+        params = {"format": "json", "query": self._sparql_query(category)}
+        return update_url_with_params(WIKIDATA_API_URL, params)
+
+    @property
+    def wikisource_api_url(self) -> str:
+        return f"https://{self.langcode}.wikisource.org/w/api.php"
+
+    @property
+    def all_wikidata_category_urls(self) -> list[str]:
+        return [self._sparql_url(c) for c in self.included_category_names]
+
+    @property
+    def excluded_categories(self) -> list[str]:
+        return [self._catformat(c) for c in self.excluded_category_names]
+
+
+# Each version of wikisource has different category names and prefixes,
+# so the pool of categories to search within and the categories to filter out
+# will have different names per wikisource version.
+# We need to explicitly exclude irrelevant categories because Wikisource does not have a unique category for books.
+# You can add more Wikisource languages here as desired.
+ws_languages = [
+    LangConfig(
+        langcode="en",
+        ol_langcode="eng",
+        category_prefix="Category",
+        included_category_names=["Validated_texts"],
+        excluded_category_names=[
+            "Subpages",
+            "Posters",
+            "Memoranda",
+            "Legislation-CAGov",
+            "Constitutional documents",
+            "National constitutions",
+            "Manuscripts",
+            "Political tracts",
+            "Proclamations",
+            "Declarations of independence",
+            "Pamphlets",
+            "Forms of government",
+            "PD-USGov",
+            "PD-CAGov",
+            "PD-UKGov",
+            "Early modern speeches",
+            "Sermons",
+            "PD-EdictGov",
+            "Film",
+        ],
+    )
+]
+
+
+def format_contributor(raw_name: str) -> str:
```

Comment:
I don't think this function is necessary. I believe it is taking a human friendly author name from Wikidata, and manipulating it to reconstruct and  human friendly author name. At best it will  be a no-op, at worst it will mix up an already good display name, depending on the black-box details of that external module. Reducing unnecessary module imports is good too.